enhancements/monitoring/hypershift-monitoring.md

```diff
@@ -0,0 +1,392 @@
+
+This is similar to how AWS, Google and so on run their managed Kubernetes service: You provide the worker nodes and they provide the control plane.
+
+The core idea of HyperShift topology is that it allows sharing infrastructure on control planes and allows the “hosted clusters” (workers) to have a lower computation and operational overhead.
+
+### Important HyperShift Assumptions
+
+* One-way communication. HyperShift management cluster can “talk to” the data plane, but not the opposite (except Kube API, Ignition and OIDC endpoint).
+* HyperShift is managed first, but form factor will be offered for self-management by 4.11.
+* There is tight budget on computation for control planes (outside of MVP).
+
+## Motivation
+
+This document enhancement proposes potential solution for Admin and Customer metric monitoring within those new topologies, since we cannot apply easily our current platform monitoring products using within cluster monitoring operator (CMO) easily (see [Alternatives](#use-pure-platform-monitoring-solution) on why).
+
+### Goals
+
+The goal is to provide a solution that maintains similar monitoring capabilities we offer with single box monitoring using CMO in HyperShift for our three [personas](#personas).
+
+> NOTE: The definition of “monitor” in the bullet list below means ability to graph, query and alert on near-real-time metrics about involved components.
+
+* HSRE can monitor management clusters and control planes running there.
+  * Ability to use AlertManager and PagerDuty workflow
+  * Ability to monitor Tugboats themselves
+  * Ability to monitor Customer control planes
+  * Ability to view fleet-level metrics and drill-downs in a single pane of glass
+  * Ability to keep metrics in-region
+* HSRE can monitor control-plane related (managed) software on guest clusters:
+  * For example Ingress/Router/CNI/CSI 
+  
+> Exact list of what is scraped on guest clusters for HSRE does not matter. All we need to know: some service monitors.
+
+* HSRE can query metrics data over SLO period (usually 28 days).
+  * Longer durations nice to have, but technically not required. There is also option that allows HSRE to persist SLO from platform to [different backend if needed](https://docs.google.com/document/d/1BJarERppgiJ8esc6d8anbJMOQ0AflFBQea-Zc9wAp0s/edit?disco=AAAAQTFsbwM).
+* HSRE have an easy way to monitor dozens of HyperShift clusters in multiple geo regions and thousands of control planes in total. This means in particular:
+  * Ability to perform PromQL across fleet metrics
+
+> NOTE(Mariusz): Alerts can be region wide only, and query in different tab per region might be good enough for now.
+
```

Comment:
We will need to have a discussion about the SRE personas and their desired dashboard points of view based on the use-cases and roles.  It would probably help to have some mockups supplied by those folks.  Don't need it in the context of this proposal.  Will need to be decided in near future.

Reply:
👍🏽